### DISCUSSION TITLE
An ingame gallery for screenshots

### DISCUSSION THREAD

**Discussion by raybeanosu** [ppy/osu#29851] [Ideas]
The idea came to me in a dream tbh.

So currently, the only way to access screenshots have been by clicking the notification on the top right, which redirects us to a screenshots folder in our file explorer. This works fine for most cases, except when someone is streaming, and they have to go through the cumbersome process of opening up said folder and either putting an image in through streaming software or sharing the window/display.

Since seeing the ingame skin editor being worked on, I thought having an ingame gallery would work out pretty well for this! It'd also look a lot cooler - though having access to the actual folder in the file explorer is also always good to have as a feature.

Here's how I imagined it myself - please let me know your thoughts on this!
![screenshot notif](https://github.com/user-attachments/assets/1b437478-d0d5-4ac0-baf4-f383bb0306d2)
_First screen - after saving a screenshot, a notification shows up in the gallery icon, prompting you to open it._

![gallery page](https://github.com/user-attachments/assets/2639f0d2-a558-41bf-8065-c3075f55a6c1)
_Second screen - the collection of screenshots taken so far, along with drop-down menus for each group. They can also be sorted! (Adding right click for extra options would be good)_

![photos](https://github.com/user-attachments/assets/67dd2525-d3fa-4136-bd33-e75cdc7a433c)
_Third screen - double click an image to open this. You can use the arrow keys on your keyboard (or your k1/k2) to go left or right, also download/export the image out of the client, or share it on the web._

**Comment by bdach** [comment]
It's an okay idea but being realistic I'm not sure it's going to happen anytime soon. I don't think it's a feature that realistically many people _need_, compared to other stuff we still have to do. 

The design is underspecified and under-documented.

- What happens if on the main view, the header with the clickable months overflows horizontally? 
- What determines whether a header of a month section has a large or small font? What determines if the thumbnail of an image is large or small?
- We can't do background blurs as you have them in the screenshots. I know you said this was "inspired by aero" but we can't do that. Blur is expensive, and blur of the entire background doubly so. Realistically the most you get is a dim.
- On the last picture the screenshot is not clearly delineated from the rest of the controls which creates a very confusing "game inside game" effect. I feel like I'm looking at a hall of mirrors when I look at that design. It's very difficult to tell what is what.

**Reply by raybeanosu** [reply]
Hello! 

Thanks for the feedback, this is my first time working on something like this so I did not know what I'm supposed to be doing in terms of specifics and documentation. I'll be working on that soon as a friend is helping me out. But I'll try to answer your questions and also update the design according to the limitations you mention.

- If it overflows horizontally, we could do what is seen on the playcount graph on the osu! website, so instead of having every month, it could show options such as every 4 months, 6 months or every year depending on the number of screenshots.
- Clicking on the month from the header, or clicking on one of the smaller thumbnails should cause the header to have a larger font. Smaller font is for the other months - likewise for the thumbnails.
- I've realised this, and i'll be replacing the blur with a dim instead.
- This is something I've only realised after posting and looking at the images with fresh eyes, though I assume I can fix that by having the menu on the very top move up temporarily. Either that, or making the gallery accessible through the Options/Settings (Ctrl/Cmd + O) like for example the Latency Certifier could work instead.

Again I realise this is not very realistic to be implemented anytime soon. I just thought it was a cool idea for a future version of lazer (as do almost anyone I've shared this idea with beforehand) and I hope it won't be impossible to work on in the future should I be able to document this better and make the actual design more feasible.

**Comment by mixalisnew** [comment]
This look intresting and this should probably be added to future plans but i agree with spaceman and its not something that we need to see anytime soon

**Reply by raybeanosu** [reply]
Thanks for the feedback! I do agree that this isn't really an important thing to be looked at anytime soon, though I'll still be working on it, just in case this is more viable in the coming months of development!

**Comment by cl8n** [comment]
looks cool! I agree with everything bdach said but this would be great to have when time allows

relevant web issue: https://github.com/ppy/osu-web/issues/3670
